File: app/src/main/java/com/stevenfrew/beatprompter/SongDisplaySettings.java
package com.stevenfrew.beatprompter;

import com.stevenfrew.beatprompter.bluetooth.ChooseSongMessage;

public class SongDisplaySettings {
    public int mOrientation;
    public int mMinFontSize;
    public int mMaxFontSize;
    public int mScreenWidth;
    public int mScreenHeight;

    SongDisplaySettings(int orientation,int minFontSize,int maxFontSize,int screenWidth,int screenHeight)
    {
        mMinFontSize=minFontSize;
        mMaxFontSize=maxFontSize;
        mOrientation=orientation;
        mScreenWidth=screenWidth;
        mScreenHeight=screenHeight;
    }

    SongDisplaySettings(ChooseSongMessage csm)
    {
        this(csm.mOrientation,csm.mMinFontSize,csm.mMaxFontSize,csm.mScreenWidth,csm.mScreenHeight);
    }
}

File: app/src/main/kotlin/com/stevenfrew/beatprompter/SongDisplaySettings.kt
package com.stevenfrew.beatprompter

import android.graphics.Rect
import com.stevenfrew.beatprompter.comm.bluetooth.message.SongChoiceInfo

class SongDisplaySettings internal constructor(var mOrientation: Int, val mMinFontSize: Float, val mMaxFontSize: Float, val mScreenSize: Rect, val mShowBeatCounter:Boolean) {
    internal constructor(choiceInfo: SongChoiceInfo) : this(choiceInfo.mOrientation, choiceInfo.mMinFontSize, choiceInfo.mMaxFontSize, choiceInfo.mScreenSize,choiceInfo.mBeatScroll||choiceInfo.mSmoothScroll)

    private val mBeatCounterHeight=
        // Top 5% of screen is used for beat counter
        if (mShowBeatCounter)
            (mScreenSize.height() / 20.0).toInt()
        else
            0
    val mBeatCounterRect = Rect(0, 0, mScreenSize.width(), mBeatCounterHeight)
    val mUsableScreenHeight=mScreenSize.height()-mBeatCounterRect.height()
}